src/test/tensor_conversion_test.cpp

```diff
@@ -626,20 +627,105 @@ TEST_F(BinaryUtilsTestKFSRawInputsContents, Positive) {
     EXPECT_EQ(std::equal(ptr, ptr + tensor.get_size(), rgb_expected_tensor), true);
 }
 
-TEST_F(BinaryUtilsTestKFSRawInputsContents, Negative_batchSizeBiggerThan1) {
+TEST_F(NativeFileInputConversionTestKFSRawInputsContents, Negative_batchSizeBiggerThan1) {
     ov::Tensor tensor;
 
     std::shared_ptr<TensorInfo> tensorInfo = std::make_shared<TensorInfo>("", ovms::Precision::U8, ovms::Shape{2, 1, 1, 3}, Layout{"NHWC"});
 
     ASSERT_EQ(convertNativeFileFormatRequestTensorToOVTensor(this->requestTensor, tensor, tensorInfo, &this->buffer), ovms::StatusCode::INVALID_BATCH_SIZE);
 }
 
-TEST_F(BinaryUtilsTestKFSRawInputsContents, Negative_emptyString) {
+TEST_F(NativeFileInputConversionTestKFSRawInputsContents, Negative_emptyString) {
     ov::Tensor tensor;
 
     std::shared_ptr<TensorInfo> tensorInfo = std::make_shared<TensorInfo>("", ovms::Precision::U8, ovms::Shape{1, 1, 1, 3}, Layout{"NHWC"});
 
     std::string empty;
     ASSERT_EQ(convertNativeFileFormatRequestTensorToOVTensor(this->requestTensor, tensor, tensorInfo, &empty), ovms::StatusCode::BYTES_CONTENTS_EMPTY);
 }
+
+template <typename TensorType>
+class StringInputsConversionTest : public ::testing::Test {
+public:
+    TensorType requestTensor;
+    void SetUp() override {}
+
+    void prepareStringTensor(tensorflow::TensorProto& tensor, std::vector<std::string> inputStrings) {
+        for (auto string : inputStrings) {
+            tensor.add_string_val(string.data(), string.size());
+        }
+        tensor.mutable_tensor_shape()->add_dim()->set_size(inputStrings.size());
+        tensor.set_dtype(tensorflow::DataType::DT_STRING);
+    }
+    void prepareStringTensor(::KFSRequest::InferInputTensor& tensor, std::vector<std::string> inputStrings) {
+        for (auto string : inputStrings) {
+            tensor.mutable_contents()->add_bytes_contents(string.data(), string.size());
+        }
+        tensor.mutable_shape()->Add(inputStrings.size());
+        tensor.set_datatype("BYTES");
+    }
+};
+
+using MyTypes = ::testing::Types<tensorflow::TensorProto, ::KFSRequest::InferInputTensor>;
+TYPED_TEST_SUITE(StringInputsConversionTest, MyTypes);
+
+static void assertOutputTensorMatchExpectations(const ov::Tensor& tensor, std::vector<std::string> inputStrings) {
+    size_t maxStringLength = 0;
+    for (const auto& input : inputStrings) {
+        maxStringLength = std::max(maxStringLength, input.length());
+    }
+    size_t width = maxStringLength + 1;
+    size_t i = 0;
+    ASSERT_EQ(width * inputStrings.size(), tensor.get_size());
+    for (const auto& input : inputStrings) {
+        for (size_t j = 0; j < input.size(); j++) {
+            ASSERT_EQ(tensor.data<uint8_t>()[i * width + j], reinterpret_cast<const uint8_t*>(input.data())[j]);
```

Comment:
You can display proper error by using stream operator:

```
ASSERT_EQ(2, 1) << "why";
```

You can use it to indicate what is exactly wrong, which `i` and `j` position cause the error